worlds/stardew_valley/options/forced_options.py

```diff
@@ -0,0 +1,48 @@
+import logging
+
+import Options as ap_options
+from . import options
+
+logger = logging.getLogger(__name__)
+
+
+def force_change_options_if_incompatible(world_options: options.StardewValleyOptions, player: int, player_name: str) -> None:
+    force_ginger_island_inclusion_when_goal_is_ginger_island_related(world_options, player, player_name)
+    force_walnutsanity_deactivation_when_ginger_island_is_excluded(world_options, player, player_name)
+    force_accessibility_to_full_when_goal_requires_all_locations(player, player_name, world_options)
+
+
+def force_ginger_island_inclusion_when_goal_is_ginger_island_related(world_options: options.StardewValleyOptions, player: int, player_name: str) -> None:
+    goal_is_walnut_hunter = world_options.goal == options.Goal.option_greatest_walnut_hunter
+    goal_is_perfection = world_options.goal == options.Goal.option_perfection
+    goal_is_island_related = goal_is_walnut_hunter or goal_is_perfection
+    ginger_island_is_excluded = world_options.exclude_ginger_island == options.ExcludeGingerIsland.option_true
+
+    if goal_is_island_related and ginger_island_is_excluded:
+        world_options.exclude_ginger_island.value = options.ExcludeGingerIsland.option_false
+        goal_name = world_options.goal.current_key
```

Comment:
Should probably use this instead:
> Goal 'greatest_walnut_hunter' requires Ginger Island.

> Goal 'Greatest Walnut Hunter' requires Ginger Island.
```suggestion
        goal_name = world_options.goal.current_option_name
```

Reply:
That looks much better, thanks!